portal-impl/src/com/liferay/portal/util/HttpImpl.java

```diff
@@ -1227,6 +1227,15 @@ public String shortenURL(String url, int count) {
 
 		StringBundler sb = new StringBundler();
 
+		String[] parts = StringUtil.split(url, CharPool.QUESTION);
+
+		if (parts.length == 2) {
```

Comment:
The example listed on the ticket was just sudo-code.  Any URL with nested redirects will have the "?"s encoded, so the "parts" variable will have a maximum length of 2.
If you look at my test case [here](https://github.com/shuyangzhou/liferay-portal/pull/4618/commits/d80409974e0a45eb66fd86b363bdbddd21439136#diff-64098f29eaff91c1f876d055a9513018R351) (line 347 if the link doesn't work), you can see I have encoded the nested redirect's "?" to "%3F".  There should only be zero or one "?" in the URL.